mini-frame.el

```diff
@@ -272,6 +278,8 @@ ALIST is passed to `window--display-buffer'."
                                   show-parameters)))
         (set-face-background 'fringe nil mini-frame-frame)))
     (modify-frame-parameters mini-frame-frame show-parameters)
+    (when mini-frame-border-color
+      (set-face-background 'internal-border mini-frame-border-color mini-frame-frame))
```

Comment:
Same here. No need to set border color every time the frame is shown.

Reply:
I tried, but it seems does not work, I am using 28.0.50 on macOS

Reply:
Mini-frame is also created here: https://github.com/muffinmad/emacs-mini-frame/blob/1fd7fcac0a5c6ba5bfdb9be00f9594fd8ffa90fd/mini-frame.el#L367

Reply:
Same result, still doesn't work . it only work when the child frame first show.

Reply:
Did you set the `mini-frame-border-color` before enabling the `mini-frame-mode`?

Reply:
yes 
```
(setq mini-frame-resize-max-height max-mini-window-height)
(setq mini-frame-completions-focus 'minibuffer)
(setq mini-frame-internal-border-color "gray70")
(when (require 'mini-frame nil t)
  (add-to-list 'mini-frame-ignore-commands 'dired-narrow)
  (setq mini-frame-show-parameters
        '((top . 0.4)
          (width . 1.0)
          (min-height .  5)
          (height . 15)
          (minibuffer-exit . t)     
          (alpha . 100)
          (left-fringe . 10 )
          (cursor-color . "Yellow")
          (background-color . "#2a3a4a")    
          (background-mode 'light)
     
          ))

  (mini-frame-mode 1))


```

Reply:
I print the value of  `(face-background 'internal-border mini-frame-frame )` before and after `(modify-frame-parameters mini-frame-frame show-parameters)`,
it seems that `modify-frame-parameters` changed that value from my `gray70` to `nil`

Reply:
You are right. Filled [bug report](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=43837) with minimal steps to reproduce this issue.

BTW there are typo in your code snippet. Replace `(background-mode 'light)` with `(background-mode . light)` ;)

Reply:
The issue is [fixed now](http://git.savannah.gnu.org/cgit/emacs.git/commit/?id=bc2b63876b669ede07c93bc5694030646d9fbd36) so we can avoid calling `set-face-background` every time the mini frame is shown.

Reply:
ok I make a force push .Will test it after emacs compiled

Reply:
it works